src/Aquifer.API/Endpoints/Resources/ResourceReferences/Get/Endpoint.cs

```diff
@@ -0,0 +1,44 @@
+using Aquifer.API.Common;
+using Aquifer.Data;
+using FastEndpoints;
+using Microsoft.EntityFrameworkCore;
+
+namespace Aquifer.API.Endpoints.Resources.ResourceReferences.Get;
+
+public class Endpoint(AquiferDbContext dbContext) : Endpoint<Request, Response>
+{
+    public override void Configure()
+    {
+        Get("/resources/resource-references");
+        Permissions(PermissionName.EditContentResourceReferences);
+    }
+
+    public override async Task HandleAsync(Request request, CancellationToken ct)
+    {
+        var query = dbContext.Resources.Where(r => r.ParentResourceId == request.ParentResourceId);
+
+        query = int.TryParse(request.ResourceIdOrExternalId, out var resourceId)
+            ? query.Where(r => r.Id == resourceId || r.ExternalId == request.ResourceIdOrExternalId)
+                .OrderBy(r => r.Id == resourceId ? 1 : 0)
+            : query.Where(r => r.ExternalId == request.ResourceIdOrExternalId);
+
+        var resource = await query
+            .Select(r => new Response
+            {
+                ResourceId = r.Id,
+                EnglishDisplayName =
```

Comment:
Is this just for display purposes? I'm wondering if we need this, since I think we just show the EnglishLabel everywhere else that displays resources for selection (resources page, dashboards, project creation, etc.)

Reply:
Yeah I was thinking display name was better for some reason but it's probably not worth handling.